packages/datasource-sql/src/introspection/introspector.ts

```diff
@@ -14,9 +14,23 @@ import { Table } from './types';
 
 export default class Introspector {
   static async introspect(sequelize: Sequelize, logger?: Logger): Promise<Table[]> {
-    const tableNames = await this.getTableNames(sequelize as SequelizeWithOptions);
-    const promises = tableNames.map(name => this.getTable(sequelize, logger, name));
-    const tables = await Promise.all(promises);
+    const tableNamesAndSchemas = await this.getTableNames(sequelize as SequelizeWithOptions);
+    const validTableNames = tableNamesAndSchemas.filter(
+      name => sequelize.getDialect() !== 'mssql' || !name.tableName.includes('.'),
+    );
+
+    if (validTableNames.length < tableNamesAndSchemas.length) {
+      const diff = tableNamesAndSchemas.filter(name => !validTableNames.includes(name));
+      logger?.(
+        'Warn',
+        `Skipping table(s): ${diff
+          .map(tableNameAndSchema => `'${tableNameAndSchema.tableName}'`)
+          .join(', ')}. MSSQL tables with dots are not supported`,
+      );
+    }
+
+    const promises = validTableNames.map(name => this.getTable(sequelize, logger, name));
+    const tables = await Promise.all(promises.filter(promise => Boolean(promise)));
```

Comment:
Filtering promises is not necessary anymore (and it did not work)

```suggestion
    const tables = await Promise.all(promises);
```